src/migrate/index.js

```diff
@@ -121,22 +138,89 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.integer('is_locked');
+    });
+  }
+
+  _getLockTableName() {
+    return this.config.tableName + '_lock';
+  }
+
+  _isLocked(trx) {
+    var tableName = this._getLockTableName();
+    return this.knex(tableName)
+      .transacting(trx)
+      .forUpdate()
+      .select('*')
+      .then(data => data[0].is_locked);
+  }
+
+  _lockMigrations(trx) {
+    var tableName = this._getLockTableName();
+    return this.knex(tableName)
+      .transacting(trx)
+      .update({ is_locked: 1 });
+  }
+
+  _getLock() {
+    return this.knex.transaction(trx => {
+      return this._isLocked(trx)
```

Comment:
What kind of problem? When the transaction of the first connection is has ran `select ... for update;` and then the transaction of the second connection tries to run the same `select ... for update;` the second connection will wait until the first transaction ends before returning result for that query.